pay-api/src/pay_api/services/eft_statements.py

```diff
@@ -0,0 +1,126 @@
+                    InvoiceModel.id == StatementInvoicesModel.invoice_id,
+                    InvoiceModel.payment_method_code == PaymentMethod.EFT.value,
+                ),
+            )
+            .filter(
+                InvoiceModel.invoice_status_code.notin_(
+                    [
+                        InvoiceStatus.CANCELLED.value,
+                        InvoiceStatus.REFUND_REQUESTED.value,
+                        InvoiceStatus.REFUNDED.value,
+                    ]
+                )
+            )
+            .group_by(StatementModel.payment_account_id)
+        )
+
+    @staticmethod
+    def get_statements_owing_as_array():
+        """Query EFT Statements with outstanding balances and pending payments."""
+        pending_payments_subquery = (
+            select(
+                func.count().label("cnt"),
+                StatementInvoicesModel.statement_id,
+                func.sum(EFTCreditInvoiceLinkModel.amount).label("pending_payment_amount"),
+            )
+            .select_from(EFTCreditInvoiceLinkModel)
+            .join(StatementInvoicesModel, StatementInvoicesModel.invoice_id == EFTCreditInvoiceLinkModel.invoice_id)
+            .where(EFTCreditInvoiceLinkModel.status_code == EFTCreditInvoiceStatus.PENDING.value)
+            .group_by(StatementInvoicesModel.statement_id)
+            .alias("pending_payments")
+        )
+
+        statements_subquery = (
+            select(
+                StatementModel.id,
+                StatementModel.payment_account_id,
+                func.sum(InvoiceModel.total - InvoiceModel.paid).label("amount_owing"),
+                func.coalesce(pending_payments_subquery.c.cnt, 0).label("pending_payments_count"),
+                func.coalesce(pending_payments_subquery.c.pending_payment_amount, 0).label("pending_payments_amount"),
+            )
+            .select_from(StatementModel)
+            .join(StatementInvoicesModel, StatementInvoicesModel.statement_id == StatementModel.id)
+            .join(InvoiceModel, InvoiceModel.id == StatementInvoicesModel.invoice_id)
+            .outerjoin(pending_payments_subquery, pending_payments_subquery.c.statement_id == StatementModel.id)
+            .where(
+                and_(
+                    StatementModel.payment_account_id == PaymentAccountModel.id,
+                    InvoiceModel.invoice_status_code.in_([InvoiceStatus.APPROVED.value, InvoiceStatus.OVERDUE.value]),
+                    InvoiceModel.payment_method_code == PaymentMethod.EFT.value,
+                )
+            )
+            .group_by(
+                StatementModel.id,
+                StatementModel.payment_account_id,
+                pending_payments_subquery.c.cnt,
+                pending_payments_subquery.c.pending_payment_amount,
+            )
+            .having(func.sum(InvoiceModel.total - InvoiceModel.paid) > 0)
+            .order_by(StatementModel.id)
+            .subquery()
+        )
+
+        json_array = func.json_agg(
```

Comment:
collapsing the all statements owing into an array of statement owing objects (this becomes dictionaries in python).

Reply:
cool